strings/strings.ts

```diff
@@ -0,0 +1,129 @@
+/**
+ * Return a number comparing two strings lexicographically.
+ *
+ * @example
+ * ```ts
+ * import { compare } from "https://deno.land/std@$STD_VERSION/strings/strings.ts";
+ *
+ * compare("a", "b"); // -1
+ * compare("a", "a"); // 0
+ * compare("b", "a"); // 1
+ * ```
+ */
+
+export function compare(a: string, b: string): number {
+  if (a === b) {
+    return 0;
+  } else if (a < b) {
+    return -1;
+  } else {
+    return 1;
+  }
+}
+
+/**
+ * Report whether a substring is within a string.
+ *
+ * @example
+ * ```ts
+ * import { contains } from "https://deno.land/std@$STD_VERSION/strings/strings.ts";
+ *
+ * contains("Hello, Deno!", "Deno"); // true
+ * ```
+ */
+
+export function contains(s: string, substr: string): boolean {
+  return s.includes(substr);
+}
+
+/**
+ * Count the number of non-overlapping instances of substr in str. Return 1 + the number
+ * of Unicode points in str if substr is empty.
+ *
+ * @example
+ * ```ts
+ * import { count } from "https://deno.land/std@$STD_VERSION/strings/strings.ts";
+ *
+ * count("cheese", "e"); // 3
+ * count("five", ""); // 5
+ * ```
+ */
+
+export function count(str: string, substr: string): number {
+  if (substr === "") return str.length + 1;
+  return str.split(substr).length - 1;
+}
+
+/**
+ * Cut str into two parts with a sep. Return before, after, found. If sep is not present in str,
+ * return str, "", false.
+ *
+ * @example
+ * ```ts
+ * import { cut } from "https://deno.land/std@$STD_VERSION/strings/strings.ts";
+ *
+ * let [before, after, found] = cut("Deno", "n"); // "De", "o", true
+ * let [before, after, found] = cut("Hello, World!", " "); // "Hello," "World!", true
+ * let [before, after, found] = cut("Fresh", "i"); // "Fresh", "", false
+ */
+
+export function cut(str: string, sep: string): [string, string, boolean] {
+  const s = str.split(sep);
+  if (s.length == 2) return [s[0], s[1], true];
+  return [str, "", false];
+}
```

Comment:
Same with this, it doesn't seem like really something people could use reliably. It just seems like a wrapper on split.

```ts
const [l, r, ...rest] = s.split(sep)
if (rest) // err
else // ok
```

Reply:
The main draw of this function over `split` is that this only splits the _first_ instance of the separator, rather than every instance.

- [Golang `Cut`](https://pkg.go.dev/strings#Cut) does seem like a very recent addition
- [Ruby `partition`](https://ruby-doc.org/3.2.0/String.html#method-i-partition) is very similar
- Doesn't seem like Voca has anything like it
- [Rust `split_once`](https://doc.rust-lang.org/std/string/struct.String.html#method.split_once)

Which also means this implementation is actually incorrect.

Rather than using `split`, (which would require a `join` after) I think we can make get more performance with something like this:

```typescript
// might be able to get a faster implementation if we use a loop instead of iterating 3 times (with indexOf, 2x slice)
const flag = s.indexOf(sep);
if (flag === -1) return [str, '', false];
return [
  str.slice(0, flag),
  str.slice(flag + sep.length),
  true,
];
```